sphinx/source/docs/user_guide/server.rst

```diff
@@ -453,101 +458,90 @@ plot):
 
-It is critical to note that **no Python code is ever executed when a CustomJS
-callback is used**. This is true even when the call back is supplied as Python
-code to be translated to JavaScript. A ``CustomJS`` callback is only executed
-inside the browser's JavaScript interpreter, and thus can only directly interact
-with JavaScript data and functions (e.g., BokehJS models).
+``CustomJS`` callbacks **never** execute Python code, not even if you convert a
+Python callback into JavaScript. ``CustomJS`` callbacks only execute inside the
+browser's JavaScript interpreter, which means that they can only interact with
+JavaScript data and functions, such as BokehJS models.
 
-Python Callbacks with Jupyter Interactors
+Python callbacks with Jupyter interactors
 '''''''''''''''''''''''''''''''''''''''''
 
-If you are working in the Jupyter notebook, it is possible to use Jupyter
-interactors to quickly create simple GUI forms automatically. Updates to the
-widgets in the GUI can trigger python callback functions that execute in
-the Jupyter Python kernel. It is often useful to have these callbacks call
-:func:`~bokeh.io.push_notebook` to push updates to displayed plots. For more
-detailed information, see :ref:`userguide_jupyter_notebook_jupyter_interactors`.
+When working with Jupyter notebooks, you can use Jupyter interactors to quickly
+create simple GUI forms. Updates to GUI widgets trigger Python callbacks that
+execute in the Python kernel of Jupyter. It is often useful to have these
+callbacks call :func:`~bokeh.io.push_notebook` to push updates to displayed
+plots. For more information, see
+:ref:`userguide_jupyter_notebook_jupyter_interactors`.
 
 .. note::
-    It is currently possible to push updates from Python to BokehJS (i.e.
-    to update plots, etc.), using :func:`~bokeh.io.push_notebook`. To add
-    two-way communication (e.g. to have a range or selection update trigger
-    a Python callback), embed a Bokeh Server in the notebook.
-    See :bokeh-tree:`examples/howto/server_embed/notebook_embed.ipynb`
+    You can push plot updates from Python to BokehJS with
+    :func:`~bokeh.io.push_notebook`. For two-way communication, embed a Bokeh
+    server in the notebook. For example, this lets range and selection updates
+    trigger Python callbacks. For further details, see
+    :bokeh-tree:`examples/howto/server_embed/notebook_embed.ipynb`
 
-Updating from Threads
+Updating from threads
 '''''''''''''''''''''
 
-If the app needs to perform blocking computation, it is possible to perform
-that work in a separate thread. However, updates to the Document must be
-scheduled via a next-tick callback. The callback
-will execute as soon as possible on the next iteration of the
-Tornado event loop, and will automatically acquire necessary locks to update the
-document state safely.
+You can make blocking computations in separate threads. However, you **have
+to** schedule document updates via a next tick callback. This callback executes
```

Comment:
I think "must" > "have to" here (in general it's probably best to avoid English phrasal verbs as much as possible, though I had not thought about that explicitly until just now cc @tcmetzger )

Reply:
I would generally try to avoid phrasal verbs because they can make reading more difficult for non-native speakers. As an alternative to "must", you could also use "required to", for example.